src/dsymbol/cache_entry.d

```diff
@@ -28,17 +28,14 @@ struct CacheEntry
 			typeid(DSymbol).destroy(symbol);
 	}
 
-	int opCmp(ref const CacheEntry other) const pure nothrow @nogc @safe
+	ptrdiff_t opCmp(ref const CacheEntry other) const pure nothrow @nogc @trusted
 	{
-		immutable int r = path > other.path;
-		if (path < other.path)
-			return -1;
-		return r;
+		return (cast(size_t) path.ptr) - (cast(size_t) other.path.ptr);
```

Comment:
try using &other.path[0] instead of .ptr. 
that way it's safe still.

Reply:
no, the codegen is awful if i do that. I don't see why this cannot be trusted.

Reply:
really, you should report a bug, because it should be the same exact code.

Reply:
I just tried it in an example: &a[0] itself is fine.
But making it @safe means that it'll insert boundschecks even in -release.

Reply:
Seriously there's no bounds check necessary here. we don't care when it's null, it's even a possible case. As for the disasm, press ASM button here : https://run.dlang.io/is/fTQdfY and you'll see.

Reply:
```asm
@safe long onlineapp.Stuff.opCmp1(ref const(onlineapp.Stuff)):
		push	RBP
		mov	RBP,RSP
		cmp	qword ptr [RDI],0
		je	L2E
		mov	RAX,8[RDI]
		cmp	qword ptr [RSI],0
		jne	L25
		mov	ESI,0Bh
		lea	RDI,_TMP0@PC32[RIP]
		call	  _d_arrayboundsp@PLT32
L25:		mov	RCX,8[RSI]
		sub	RAX,RCX
		pop	RBP
		ret
L2E:		mov	ESI,0Bh
		lea	RDI,_TMP0@PC32[RIP]
		call	  _d_arrayboundsp@PLT32
		pop	RBP
		ret
```

VS

```asm
push	RBP
mov	RBP,RSP
mov	RAX,8[RDI]
sub	RAX,8[RSI]
pop	RBP
ret
```

Reply:
@UplinkCoder, i've asked to Hackerpilot to make the final decision but i think that you miss the fact that `.ptr` is for a string interning entry, which is why i think the fast solution (subtract) can be trusted.

Reply:
&array[0] is terrible, since it breaks generic code due to the possibility of dereferencing a null pointer. See https://run.dlang.io/is/n8T9Ul for a blatant example. The C++ standard libraries introduced data() accessors to applicable containers, and &array[0] caused a few bugs during Quantum Break development before we moved them over to array.data() calls.

Encouraging &array[0] really isn't a good idea.

Reply:
The whole point of doing the string interning is that we can treat the pointers as integers for the purpose of comparing them.